cpu/rpx0xx/include/pio/pio.h

```diff
@@ -0,0 +1,1166 @@
+ * @param[in]       source          Where to move from
+ *
+ * @return  MOV instruction
+ */
+static inline pio_instr_t pio_inst_mov(pio_inst_mov_dst_t destination,
+                                       pio_inst_mov_op_t operation,
+                                       pio_inst_mov_src_t source)
+{
+    return PIO_INST_MOV |
+           ((unsigned)destination << PIO_INST_MOV_DESTINATION_SHIFT) |
+           ((unsigned)operation << PIO_INST_MOV_OP_SHIFT) |
+           ((unsigned)source << PIO_INST_MOV_SOURCE_SHIFT);
+}
+/**
+ * @brief   Construct an IRQ instruction
+ *
+ * @param[in]       clear           Clear IRQ instead to raise it
+ * @param[in]       wait            Wait until IRQ is cleared if raised
+ * @param[in]       relative        IRQ index is relative to SM number
+ * @param[in]       index           IRQ index
+ *
+ * @return  IRQ instruction
+ */
+static inline pio_instr_t pio_inst_irq(bool clear,
+                                       bool wait,
+                                       bool relative,
+                                       unsigned index)
+{
+    return PIO_INST_IRQ |
+           (!!clear << PIO_INST_IRQ_CLR_SHIFT) |
+           (!!wait << PIO_INST_IRQ_WAIT_SHIFT) |
+           ((!!relative << 4) | index);
+}
+/**
+ * @brief   Construct a SET instruction
+ *
+ * @param[in]       destination     Set destination
+ * @param[in]       data            Data to set
+ *
+ * @return  SET instruction
+ */
+static inline pio_instr_t pio_inst_set(pio_inst_set_dst_t destination,
+                                       unsigned data)
+{
+    return PIO_INST_SET |
+           ((unsigned)destination << PIO_INST_SET_DESTINATION_SHIFT) |
+           data;
+}
+
+/**
+ * @brief   Encode the delay/sideset instruction field
+ *
+ * @param[in]       sideset         Sideset value to write to sideset pins
+ * @param[in]       sideset_count   Number of sideset pins
+ * @param[in]       sideset_opt     If sideset is optional
+ * @param[in]       delay           Delay cycles
+ *
+ * @return  Delay/sideset encoding that can be added to an instruction
+ */
+static inline pio_instr_t pio_inst_delay_sideset(unsigned sideset,
+                                                 unsigned sideset_count,
+                                                 bool sideset_opt,
+                                                 unsigned delay)
+{
+    return ((!!sideset_opt) << 12) |
+           (sideset << (13 - (!!sideset_opt + sideset_count))) |
+           (delay << 8);
+}
+/** @} */
+
+/**
+ * @brief   Internal state machine registers
+ */
+typedef struct pio_sm_ctrl_regs {
+    volatile uint32_t clkdiv;       /**< SMx_CLKDIV */
+    volatile uint32_t execctrl;     /**< SMx_EXECCTRL*/
+    volatile uint32_t shiftctrl;    /**< SMx_SHIFTCTRL */
+    const volatile uint32_t addr;   /**< SMx_ADDR */
+    volatile uint32_t instr;        /**< SMx_INSTR */
+    volatile uint32_t pinctrl;      /**< SMx_PINCTRL */
+} pio_sm_ctrl_regs_t;
```

Comment:
Is this provided by the new header files from #19416?

Reply:
It is a struct that maps the registers for PIO configuration to generalize for PIO0 or PIO1 and their state machine 1 to 4.
The current code works witch the changes of #19416. 

```C
typedef struct { 
  ...
  __IOM uint32_t  SM0_CLKDIV;                   /*!< (@ 0x000000C8) Clock divisor register for state machine 0
                                                                    Frequency = clock freq / (CLKDIV_INT + CLKDIV_FRAC
                                                                    / 256)                                                     */
  __IOM uint32_t  SM0_EXECCTRL;                 /*!< (@ 0x000000CC) Execution/behavioural settings for state machine
                                                                    0                                                          */
  __IOM uint32_t  SM0_SHIFTCTRL;                /*!< (@ 0x000000D0) Control behaviour of the input/output shift registers
                                                                    for state machine 0                                        */
  __IOM uint32_t  SM0_ADDR;                     /*!< (@ 0x000000D4) Current instruction address of state machine
                                                                    0                                                          */
  __IOM uint32_t  SM0_INSTR;                    /*!< (@ 0x000000D8) Read to see the instruction currently addressed
                                                                    by state machine 0's program counter
                                                                    Write to execute an instruction immediately
                                                                    (including jumps) and then resume execution.               */
  __IOM uint32_t  SM0_PINCTRL;                  /*!< (@ 0x000000DC) State machine pin control                                  */
  ...
} PIO0_Type;
```